apps/gapilotbuddy/ga_pilot_buddy.star

```diff
@@ -0,0 +1,201 @@
+"""
+Applet: GA Pilot Buddy
+Summary: Local flight rules and wx
+Description: See local aerodrome flight rules and current abbreviated METAR information.
+Author: icdevin
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("humanize.star", "humanize")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+load("time.star", "time")
+
+AVWX_TOKEN = """
+AV6+xWcEMClLASnRjwefBSKdSfw7sY2iH8i5AqAiR07g6hb1tptkeFwK31hnx4Y0tdUUBNZ+4zFkhz
+TgMw38WzY+XrFGx2TVG0Aif1XERXmutNCIc9PMLuW4vv2penHt100RwCWglurpqNf0T8ZPLvEUg816
+PG3iHrOD00eIMlxaI8LcEABZYgmutmPWxUXmyg==
+"""
+DEFAULT_LOCATION = """
+{
+    "lat": "33.6295968",
+    "lng": "-117.8862308",
+    "description": "Newport Beach, CA, USA",
+    "locality": "Newport Beach",
+    "place_id": "ChIJ3whWdFnf3IARUV7GZxqUpjs",
+    "timezone": "America/Los_Angeles"
+}
+"""
+DEFAULT_FLIGHT_RULES_COLOR = "#C3C3C3"
+ERROR_ICON = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAAwAAAAMCAYAAABWdVznAAAATElEQVQoU2P8L8L8n/HNX0YGAgCmDq
+yQkCZkebjJuDShi6M4BV0SmyEYbocpwmUjVs/i8xNlNpDkB5JCiaR4IKQYlgBQYppQskCWBwCgNlQN
+phkLigAAAABJRU5ErkJggg==
+""")
+FLIGHT_RULES_COLOR_MAP = {
+    "VFR": "#01CF00",
+    "MVFR": "#0061E7",
+    "IFR": "#EB0000",
+    "LIFR": "#D300D3",
+}
+
+def get_avwx_headers(config):
+    return {
+        "Authorization": "Token {}".format(secret.decrypt(AVWX_TOKEN) or config.get("avwx_token")),
+    }
+
+def get_nearby_aerodromes(location, config):
+    # Truncates the precise lat/lng for data privacy since this will be sent out
+    # to a third party API
+    lat = humanize.float("#.#", float(location["lat"]))
+    lng = humanize.float("#.#", float(location["lng"]))
+    str_geo = "{},{}".format(lat, lng)
+
+    aerodromes = cache.get(str_geo)
+    if aerodromes == None:
+        url = "https://avwx.rest/api/station/near/{}".format(str_geo)
```

Comment:
Great work making sure your calls are limited! It looks like you'll get 4K calls per day assuming you're on the free plan. I see you're also aggressively caching and truncating the lat/lng. One decimal place is [11.1km/6.9mi](https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude) so there will certainly be only a few calls per week for a major metropolitan area like New York or Los Angeles. If we ever get so many users where this app is doing 4k calls per day in the current form we'd also be in a position to just pay for it 😄

Reply:
Thanks! It would've ended up being pretty noisy otherwise for sure. One digit is better for caching but often there's many airports in a tight space so it may need to be increased later if it starts looking wrong. In any case, luckily airports don't open or close often at all so it can be heavily cached. 😎 

That's very generous of you. For now I've got overages turned on until it goes over the limit, but I'll let you know if things get crazy. At some point it'll probably be cheaper to pay for the tier anyway because their first paid tier has a "Multiple Reports" call so I can group three requests into one. Food for thought lol.